spectrum_fundamentals/metrics/similarity.py

```
@@ -56,19 +56,17 @@ def spectral_angle(
                                      position >= peptide length), array of length 174
         :param predicted_intensities: predicted intensities, see observed_intensities for details, array of length 174
         :param charge: to filter by the peak charges, 0 means everything
+        :param xl: whether operating on cleavable crosslinked or linear peptides
         :return: SA values
         """
         if charge != 0:
-            if charge == 1:
-                boolean_array = constants.SINGLE_CHARGED_MASK
-            elif charge == 2:
-                boolean_array = constants.DOUBLE_CHARGED_MASK
-            elif charge == 3:
-                boolean_array = constants.TRIPLE_CHARGED_MASK
-            elif charge == 4:
-                boolean_array = constants.B_ION_MASK
+            if xl:
+                masks = constants.MASK_DICT_XL
+                default_value = constants.B_ION_MASK_XL
```

remove default values and make ia one liner: `masks = constants.MASK_DICT_XL if xl else constants.MASK_DICT`